cardano-api/src/Cardano/Api/SpecialByron.hs

```diff
@@ -0,0 +1,185 @@
+
+    metaData :: M.Map SystemTag InstallerHash
+    metaData = M.singleton sysTag insHash
+
+    noPassSigningKey :: SafeSigner
+    noPassSigningKey = noPassSafeSigner sKey
+
+    protocolParamsUpdate :: ProtocolParametersUpdate
+    protocolParamsUpdate = createProtocolParametersUpdate
+                         emptyProtocolParametersUpdate paramsToUpdate
+
+createProtocolParametersUpdate
+  :: ProtocolParametersUpdate
+  -> [ParametersToUpdate]
+  -> ProtocolParametersUpdate
+createProtocolParametersUpdate = go
+ where
+   go i [] = i
+   go i (paramToUpdate : rest) =
+     case paramToUpdate of
+       ScriptVersion val -> go i{ppuScriptVersion = Just val} rest
+       SlotDuration val -> go i{ppuSlotDuration = Just val} rest
+       MaxBlockSize val -> go i{ppuMaxBlockSize = Just val} rest
+       MaxHeaderSize val -> go i{ppuMaxHeaderSize = Just val} rest
+       MaxTxSize val -> go i{ppuMaxTxSize = Just val} rest
+       MaxProposalSize val -> go i{ppuMaxProposalSize = Just val} rest
+       MpcThd val -> go i{ppuMpcThd = Just val} rest
+       HeavyDelThd val -> go i{ppuHeavyDelThd = Just val} rest
+       UpdateVoteThd val -> go i{ppuUpdateVoteThd = Just val} rest
+       UpdateProposalThd val -> go i{ppuUpdateProposalThd = Just val} rest
+       UpdateProposalTTL val -> go i{ppuUpdateProposalTTL = Just val} rest
+       SoftforkRuleParam val -> go i{ppuSoftforkRule = Just val} rest
+       TxFeePolicy val -> go i{ppuTxFeePolicy = Just val} rest
+       UnlockStakeEpoch val -> go i{ppuUnlockStakeEpoch = Just val} rest
+
+emptyProtocolParametersUpdate :: ProtocolParametersUpdate
+emptyProtocolParametersUpdate =
+  ProtocolParametersUpdate
+    { ppuScriptVersion = Nothing
+    , ppuSlotDuration = Nothing
+    , ppuMaxBlockSize = Nothing
+    , ppuMaxHeaderSize = Nothing
+    , ppuMaxTxSize = Nothing
+    , ppuMaxProposalSize = Nothing
+    , ppuMpcThd = Nothing
+    , ppuHeavyDelThd = Nothing
+    , ppuUpdateVoteThd = Nothing
+    , ppuUpdateProposalThd = Nothing
+    , ppuUpdateProposalTTL = Nothing
+    , ppuSoftforkRule = Nothing
+    , ppuTxFeePolicy = Nothing
+    , ppuUnlockStakeEpoch = Nothing
+    }
+
+data ParametersToUpdate
+  = ScriptVersion Word16
+  | SlotDuration Natural
+  | MaxBlockSize Natural
+  | MaxHeaderSize Natural
+  | MaxTxSize Natural
+  | MaxProposalSize Natural
+  | MpcThd LovelacePortion
+  | HeavyDelThd LovelacePortion
+  | UpdateVoteThd LovelacePortion
+  -- ^ UpdateVoteThd: This represents the minimum percentage
+  -- of the total number of genesis keys that have to endorse
+  -- a protocol version in order for that version to be adopted.
+  | UpdateProposalThd LovelacePortion
+  -- ^ UpdateProposalTTL: If after the number of slots specified the proposal
+  -- does not reach majority of approvals, the proposal is simply discarded.
+  | UpdateProposalTTL SlotNumber
+  | SoftforkRuleParam SoftforkRule
+  | TxFeePolicy TxFeePolicy
+  | UnlockStakeEpoch EpochNumber
```

Comment:
I see what you're doing here, and it is actually nicer to use.

However, I'd still recommend that we follow the style we use for the Shelley protocol update proposals (see `ProtocolParametersUpdate`), and change this to be `ByronProtocolParametersUpdate` that is just a big record of Maybes.

The main reason to do it this way, apart from consistency with Shelley, is that it means we will be able to use the same type for reading the chain later, and seeing the existing Byron-era update proposals.

That said, I see this is the way it was being done in the CLI before, and it's not terrible. We can leave changing this to later instead. Just leave a TODO about it, and note that we will need this once we support using these types for reading the chain, as well as just creating (non-)transactions to post to the chain.